oauth2client/client.py

```diff
@@ -598,19 +598,9 @@ def new_request(uri, method='GET', body=None, headers=None,
 
             # Clone and modify the request headers to add the appropriate
             # Authorization header.
-            if headers is None:
-                headers = {}
-            else:
-                headers = dict(headers)
+            headers = self._initialize_headers(headers)
```

Comment:
The semantics here are slightly changed, aren't they? In the code on the left side of the diff, the adjusted user-agent value is not included in the headers passed to `self.apply`, but in the code on the right it is. Is this a concern?